it/api/services/AuthStub.scala

```diff
@@ -0,0 +1,163 @@
+      .withRequestBody(
+        followUpAgentEnrolmentsRequest(EnrolmentsAuthService.primaryAgentAuthPredicate(mtdId = "1234567890"))
+      )
+      .thenReturn(status = OK, body = successfulAuthResponse("Agent", primaryAgentEnrolmentResponse))
+
+  def unauthorisedForPrimaryAgentEnrolment(): StubMapping =
+    when(method = POST, uri = authoriseUri)
+      .withRequestBody(
+        followUpAgentEnrolmentsRequest(EnrolmentsAuthService.primaryAgentAuthPredicate(mtdId = "1234567890"))
+      )
+      .thenReturn(status = UNAUTHORIZED, headers = Map("WWW-Authenticate" -> """MDTP detail="FailedRelationship""""))
+
+  def authorisedWithSupportingAgentEnrolment(): StubMapping =
+    when(method = POST, uri = authoriseUri)
+      .withRequestBody(
+        followUpAgentEnrolmentsRequest(EnrolmentsAuthService.supportingAgentAuthPredicate(mtdId = "1234567890"))
+      )
+      .thenReturn(status = OK, body = successfulAuthResponse("Agent", supportingAgentEnrolmentResponse))
+
+  def unauthorisedForSupportingAgentEnrolment(): StubMapping =
+    when(method = POST, uri = authoriseUri)
+      .withRequestBody(
+        followUpAgentEnrolmentsRequest(EnrolmentsAuthService.supportingAgentAuthPredicate(mtdId = "1234567890"))
+      )
+      .thenReturn(status = UNAUTHORIZED, headers = Map("WWW-Authenticate" -> """MDTP detail="FailedRelationship""""))
+
+  def unauthorisedNotLoggedIn(): StubMapping =
+    // Note that MissingBearerToken extends NoActiveSession
+    when(method = POST, uri = authoriseUri)
+      .thenReturn(status = UNAUTHORIZED, headers = Map("WWW-Authenticate" -> """MDTP detail="MissingBearerToken""""))
+
+  def unauthorisedOther(): StubMapping =
+    when(method = POST, uri = authoriseUri)
+      .thenReturn(status = UNAUTHORIZED, headers = Map("WWW-Authenticate" -> """MDTP detail="InvalidBearerToken""""))
+
+  private def authorisedWith(affinityGroup: String): StubMapping =
+    when(method = POST, uri = authoriseUri)
+      .withRequestBody(initialRetrievalsRequestBody)
+      .thenReturn(
+        status = OK,
+        body = successfulAuthResponse(
+          affinityGroup = affinityGroup,
+          enrolments = initialAgentEnrolmentResponse
+        ))
+
+  private def successfulAuthResponse(affinityGroup: String, enrolments: Seq[JsObject]): JsObject =
+    Json.obj(
+      "affinityGroup"        -> affinityGroup,
+      "authorisedEnrolments" -> enrolments
+    )
+
+  private def followUpAgentEnrolmentsRequest(enrolment: Enrolment): JsValue = {
+    JsObject(
+      Map(
+        "authorise" -> JsArray(List(enrolment.toJson)),
+        "retrieve"  -> JsArray.empty
+      ))
+  }
+
+  private val initialAgentEnrolmentResponse = List(
+    Json.obj(
+      "key" -> "HMRC-AS-AGENT",
+      "identifiers" -> Json.arr(
+        Json.obj(
+          "key"   -> "AgentReferenceNumber",
+          "value" -> "1234567890"
+        )
+      )
+    )
+  )
+
+  private val primaryAgentEnrolmentResponse = List(
```

Comment:
I think this needs to be updated to return  an empty JSON object `Json.obj()` as both Primary/Main agent and Secondary/Supporting agent auth calls won't return any JSON data back but just Http status 200 or 401 etc and empty JSON object.